File: BackEnd/src/main/java/com/example/backend/model/Seat.java
package com.example.backend.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.*;

@Entity
@SequenceGenerator(
        name = "SQ_SEAT_GENERATOR"
        , sequenceName = "SQ_SEAT"
        , initialValue = 1
        , allocationSize = 1
)
@Table(name = "TB_SEAT")
@Getter
@Setter
@NoArgsConstructor
@DynamicInsert
@DynamicUpdate
// soft delete
//@Where(clause = "DELETE_YN = 'N'")
//@SQLDelete(sql = "UPDATE TB_SEAT SET DELETE_YN = 'Y', DELETE_TIME=TO_CHAR(SYSDATE, 'YYYY-MM-DD HH24:MI:SS') WHERE SEATPOSITION = ?")
public class Seat {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SQ_SEAT_GENERATOR")
    @Column(columnDefinition = "NUMBER")
    private Long sno;

    @Column(columnDefinition = "VARCHAR2(4)")
    private String seatposition;

    @Column(columnDefinition = "VARCHAR2(1)", name = "SEATYN")
    private String seatyn;

    @Column(columnDefinition = "NUMBER")
    private Long scno;


    public Seat(String seatposition, String seatyn, Long scno) {
        this.seatposition = seatposition;
        this.seatyn = seatyn;
        this.scno = scno;
    }
}

File: frontEnd/src/model/Seat.js
export default class Seat {
    // js 생성자 : constructor(매개변수...) {}
    constructor(
        sno, // 자동생성 시퀀스 번호
        seatposition, // 좌석 이름
        seatyn, // 좌석 예약 여부(Y/n)
        scno // 스케쥴 넘버
    ) {
        this.sno = sno;
        this.seatposition = seatposition;
        this.seatyn = seatyn;
        this.scno = scno;
    }
}